mopidy/m3u/playlists.py

```diff
@@ -89,8 +94,10 @@ def _write_m3u_extinf(self, file_handle, track):
         file_handle.write('#EXTINF:' + str(runtime) + ',' + title + '\n')
 
     def _sanitize_m3u_name(self, name, encoding=sys.getfilesystemencoding()):
+        name = self._invalid_filename_chars.sub('|', name.strip())
+        # make sure we end up with a valid path segment
         name = name.encode(encoding, errors='replace')
-        name = os.path.basename(name)
+        name = os.path.basename(name)  # paranoia?
```

Comment:
Should be a noop when slashes are already replaced, but I feel safer leaving it in.